pkg/volumeclaimbinder/types.go

```diff
@@ -139,3 +145,65 @@ func matchStorageCapacity(pvA, pvB *api.PersistentVolume) bool {
 	bSize := bQty.Value()
 	return aSize <= bSize
 }
+
+// allPossibleMatchingAccessModes returns an array of AccessMode arrays that can satisfy a user's requested modes.
+// There are, essentially, 3 ways to request access modes:
+//
+//  RWO -- the most basic form of access (all storage volumes)
+//  RWO+ROX -- 1 writer or many read-only readers (GCEPersistentDisk)
+//  RWO+ROX+RWX -- many writers (NFS)
+//
+// A user's request can be satisfied with more modes, but never fewer.
+// This func returns modes with ascending levels of modes to give the user what is closest to what they actually asked for.
```

Comment:
Some examples of input -> output would also help.  I don't grok slice-of-slice-of mode as the return